src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -88,6 +88,36 @@ ov::Tensor tensor_from_numpy(py::array& array, bool shared_memory) {
     return tensor;
 }
 
+ov::PartialShape partial_shape_from_list(const py::list& shape) {
+    using value_type = ov::Dimension::value_type;
+    std::vector<ov::Dimension> dims;
+    for (py::handle dim : shape) {
+        if (py::isinstance<py::int_>(dim)) {
+            dims.push_back(ov::Dimension(dim.cast<value_type>()));
+        } else if (py::isinstance<ov::Dimension>(dim)) {
+            dims.push_back(dim.cast<ov::Dimension>());
+        } else if (py::isinstance<py::list>(dim) || py::isinstance<py::tuple>(dim)) {
+            py::list bounded_dim = dim.cast<py::list>();
+            py::type_error error("Incorrect values in shape. Can't create bounded dimension from " +
+                                 std::string(dim.str()) + ".");
+            if (bounded_dim.size() != 2) {
+                throw error;
+            }
+            if (!(py::isinstance<py::int_>(bounded_dim[0]) && py::isinstance<py::int_>(bounded_dim[1]))) {
+                throw error;
+            }
+            dims.push_back(ov::Dimension(bounded_dim[0].cast<value_type>(), bounded_dim[1].cast<value_type>()));
+        } else if (py::isinstance(dim, py::module::import("builtins").attr("range"))) {
+            dims.push_back(
+                ov::Dimension(dim.attr("start").cast<value_type>(), dim.attr("stop").cast<value_type>() - 1));
```

Comment:
Range looks native for Python, but the only thing that concerns me is that we have to always +1 to upper bound because the upper bound is not included for Python range. In our native form with `[l, u]`, `u` is included. So range(10) is not the same as [0, 10], but it is [0, 9]. So we have an inconsistency between our native range syntax and python range. It makes picture more complex for the user. @ilya-lavrenov, @ilyachur, @jiwaszki and others, can you vote, should we really introduce Python range support in this case? I understand that I've coined this range story, but it's not too late to change.

Reply:
Also one note hear to the side of not supporting it that python range has `step` attribute which is not compatible with dynamic dimension logic

Reply:
range is removed